.github/workflows/pkgci_sharktank_regression.yml

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: PkgCI Sharktank Regression Tests
+on:
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+
+jobs:
+  sharktank_regresion_tests:
+    name: "sharktank_regresion_tests :: ${{ matrix.name }}"
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # CPU
+          - name: cpu_task_model
+            target: target_cpu
+            gpu: none
+            runs-on: ubuntu-24.04
+          - name: cpu_llvm_task_regression
+            models-config-file: models_cpu_llvm_task.json
+            backend: cpu
+            runs-on:
+              - self-hosted # must come first
+              - persistent-cache
+              - Linux
+              - X64
+
+          # AMD GPU
+          - name: rocm_hip_model
+            target: target_hip
+            gpu: gfx1100
+            runs-on: nodai-amdgpu-w7900-x86-64
+          - name: amdgpu_rocm_mi250_gfx90a_regression
+            rocm-chip: gfx90a
+            backend: rocm
+            sku: mi250
+            runs-on: nodai-amdgpu-mi250-x86-64
+          - name: amdgpu_rocm_mi300_gfx942_regression
+            rocm-chip: gfx942
+            backend: rocm
+            sku: mi300
+            runs-on: nodai-amdgpu-mi300-x86-64
+          - name: amdgpu_rocm_mi308_gfx942_regression
+            rocm-chip: gfx942
+            backend: rocm
+            sku: mi308
+            runs-on: nodai-amdgpu-mi308-x86-64
+
+    env:
+      VENV_DIR: ${{ github.workspace }}/venv
+      PACKAGE_DOWNLOAD_DIR: ${{ github.workspace }}/.packages
+      TEST_OUTPUT_ARTIFACTS: ${{ github.workspace }}/model_output_artifacts
+    steps:
+      - name: Checking out IREE repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: false
+
+      - uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
+        with:
+          # Must match the subset of versions built in pkgci_build_packages.
+          python-version: "3.11"
+
+      - uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: linux_x86_64_release_packages
+          path: ${{ env.PACKAGE_DOWNLOAD_DIR }}
+
+      - name: Setup Python virtual environment
+        run: |
+          ./build_tools/pkgci/setup_venv.py ${VENV_DIR} \
+            --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
+            --fetch-gh-workflow=${{ inputs.artifact_run_id }}
+
+      - name: "Running Sharktank regression composite action"
+        uses: iree-org/iree-test-suites/sharktank_models@main
```

Comment:
Pin actions to specific commit hashes. For actions we _don't_ control, that prevents actions from being updated without our notice. For actions we _do_ control, that lets us make cross-repository changes without needing to precisely coordinate merging.

Each workflow run should be repeatable. Having unpinned dependencies can lead to a re-run of an action having different behavior.